unified-installer/astra-unified-installer.sh

```diff
@@ -99,15 +99,38 @@ readonly __FATAL=50
 #     captured_stdout="$(curl -sS https://bad-url.com 2> "$_ERR_FILE")"
 #     captured_stderr="$(get_captured_err)"
 readonly __ERR_FILE="tmp_last_captured_error.txt"
+
+# __TMP_ENV is used to store the user's env vars so that we can then re-apply them after having sourced
+# their config file. This allows us to make command line vars take precedence over what's in the config file.
+readonly __TMP_ENV="tmp.env"
```

Comment:
As the comment mentions, this is used to temporarily hold the user's environment variables (not from a config file, but the ones inherited by the script from the parent shell). The order of operations are roughly:

1. Store inherited environment
2. Apply user's config file
3. Apply stored inherited environment

This allows us to reverse the priority of the variables, making CLI args take precedence over the config file. This is something that was requested by PMs, though the only reason it hadn't been done in the first place regardless was the challenge of doing it in bash being surprisingly not-so-easy to solve, as you'll see later (I'll point out the relevant code in a comment).